src/test/scala/edu/ie3/simona/model/participant/HpModelSpec.scala

```diff
@@ -232,55 +232,209 @@ class HpModelSpec
+      "deliver correct flexibility options" in {
+        val testCases
+            : TableFor3[ThermalGridState, HpState, (Double, Double, Double)] =
+          Table(
+            ("thermalState", "lastState", "expectedValues"),
+            // House is below lower temperature boundary
+            (
+              ThermalGridState(
+                Some(ThermalHouseState(0L, Celsius(15), Kilowatts(0))),
+                Some(ThermalStorageState(0L, KilowattHours(20), Kilowatts(0))),
+              ),
+              HpState(
+                isRunning = false,
+                0,
+                Some(hpData.ambientTemperature),
+                Kilowatts(0.0),
+                Kilowatts(0.0),
+                ThermalGridState(
+                  Some(ThermalHouseState(0L, Celsius(15), Kilowatts(0))),
+                  Some(
+                    ThermalStorageState(0L, KilowattHours(20), Kilowatts(0))
+                  ),
+                ),
+                None,
+              ),
+              (95.0, 95.0, 95.0),
             ),
-            Some(
-              ThermalStorageState(
-                0L,
-                KilowattHours(20),
-                Kilowatts(0),
-              )
+            // House is between target temperature and lower temperature boundary, Hp actually running
+            (
+              ThermalGridState(
+                Some(ThermalHouseState(0L, Celsius(19), Kilowatts(0))),
+                Some(ThermalStorageState(0L, KilowattHours(20), Kilowatts(0))),
+              ),
+              HpState(
+                isRunning = true,
+                0,
+                Some(hpData.ambientTemperature),
+                Kilowatts(95.0),
+                Kilowatts(80.0),
+                ThermalGridState(
+                  Some(ThermalHouseState(0L, Celsius(19), Kilowatts(80))),
+                  Some(
+                    ThermalStorageState(0L, KilowattHours(20), Kilowatts(0))
+                  ),
+                ),
+                None,
+              ),
+              (95.0, 0.0, 95.0),
+            ),
+
+            // House is between target temperature and lower temperature boundary, Hp actually not running
+            (
+              ThermalGridState(
+                Some(ThermalHouseState(0L, Celsius(19), Kilowatts(0))),
+                Some(ThermalStorageState(0L, KilowattHours(20), Kilowatts(0))),
+              ),
+              HpState(
+                isRunning = false,
+                0,
+                Some(hpData.ambientTemperature),
+                Kilowatts(0.0),
+                Kilowatts(0.0),
+                ThermalGridState(
+                  Some(ThermalHouseState(0L, Celsius(19), Kilowatts(0))),
+                  Some(
+                    ThermalStorageState(0L, KilowattHours(20), Kilowatts(0))
+                  ),
+                ),
+                None,
+              ),
+              (0.0, 0.0, 95.0),
+            ),
```

Comment:
To add to my last comment, this would be an example. Why does it matter whether the heat pump has been running before? Aren't we deciding on that at this point in time anew?

Reply:
The current state depends on the last power, because that is how we distinguish what the reference behaviour will be in this case. For example: House is between lower and target temperature. If the HP was active in the last state, the house was in the heating phase, so our reference behaviour will still be 'active HP'. If it was not active in the last state, the house is cooling, our reference behaviour should be non-active hp until the house reaches the lower temp boundary.